internal/rbd/snapshot.go

```diff
@@ -156,11 +156,13 @@ func (rbdSnap *rbdSnapshot) toVolume() *rbdVolume {
 }
 
 func (rbdSnap *rbdSnapshot) ToCSI(ctx context.Context) (*csi.Snapshot, error) {
+	//nolint:gocritic // intentionally commented out the second case
 	switch {
 	case rbdSnap.VolID == "":
 		return nil, fmt.Errorf("%q does not have a volume-id set", rbdSnap)
-	case rbdSnap.SourceVolumeID == "":
-		return nil, fmt.Errorf("%q does not have a source-volume-id set", rbdSnap)
+		// FIXME: Re-enable this once the snapshot has a source volume ID set.
+		// case rbdSnap.SourceVolumeID == "":
+		// 	return nil, fmt.Errorf("%q does not have a source-volume-id set", rbdSnap)
```

Comment:
This is not the right fix beause this is used for CreateSnapshot as well. If required i would suggest to discard the error in List operation so that we will not have any problem for existing work flow.

Reply:
Please see https://github.com/red-hat-storage/ceph-csi/pull/538#discussion_r2115405758

Reply:
As i mentioned above this will have side effect as the function is used in other RPC. i would suggest to fix this only in List RPC so that we will not impact any other RPC. @nixpanic Is this fine for you?